--- S2EExecutionStateRegisters.h.orig	2022-06-14 17:18:07.747867598 +0200
+++ S2EExecutionStateRegisters.h	2022-06-14 17:19:53.660827194 +0200
@@ -330,6 +330,9 @@
     /// \param pc the new value of the program counter
     ///
     void setPc(uint64_t pc);
+
+    /// Check if a register is currently concrete or symbolic.
+    bool getRegType(unsigned offset, unsigned size, bool *isConcrete);
 };
 } // namespace s2e
 
